### PR TITLE
Highlight selected view and clear selection after switch trace

### LINKED ISSUE #540: Selected View is not highlighted when clicking on a view in the list of available views
After opening a trace and clicking on a view in the list of available views, the selected view is not highlighted. It used to show always the last selected view in the list. Resizing the window will trigger the a rendering and then the last view is selected.

views-missing-selection

### PR DESCRIPTION
**Pull request by ywang001**
fixes #540

1) Whenever you click on one of the available views, it is highlighted immediately even if you do not resize the sidebar.
2) Whenever you switch to another opened trace, the view selection will be cleared. For example, you have wget-second-call opened and CPU Usage selected and highlighted. If you switch from wget-second-call to wget-first-call, none of the available views will be selected or highlighted.

Signed-off-by: yining wang <yining.wang@ericsson.com

### CODE REVIEW

**Review by bhufmann — DISMISSED** [review]
Looks good to me. Thanks for the contribution.

**Review by IbrahimFradj — CHANGES_REQUESTED** [review]
LGTM, i think you have just to squash the 2 commit into 1. Thank you for your contribution.

**Comment by mirsky-work** [comment]
> LGTM, i think you have just to squash the 2 commit into 1. Thank you for your contribution.

The 2 commits seem unrelated, so IMO should remain separated. But @ywang001, please format the commit messages: short title in imperative form, starts with a capital letter

File: packages/react-components/src/trace-explorer/trace-explorer-views-widget.tsx [hunk 1]
```diff
@@ -90,6 +90,7 @@ export class ReactAvailableViewsWidget extends React.Component<ReactAvailableVie
             traceContainerClassName = traceContainerClassName + ' theia-mod-selected';
         }
         return <div className={traceContainerClassName}
+            title={outputName + ': ' + outputDescription}
```
Comment (PatrickTasse) [inline packages/react-components/src/trace-explorer/trace-explorer-views-widget.tsx]: What is the effect of this line?

File: packages/react-components/src/trace-explorer/trace-explorer-views-widget.tsx [hunk 2]
```diff
@@ -90,6 +90,7 @@ export class ReactAvailableViewsWidget extends React.Component<ReactAvailableVie
...
             id={`${traceContainerClassName}-${props.index}`}
             key={props.key}
             style={props.style}
```
Comment (PatrickTasse) [inline packages/react-components/src/trace-explorer/trace-explorer-views-widget.tsx]: Is there supposed to be a tooltip when hovering over a text with the ellipsis? I don't see any.

File: packages/react-components/src/trace-explorer/trace-explorer-views-widget.tsx [hunk 3]
```diff
@@ -86,11 +86,10 @@ export class ReactAvailableViewsWidget extends React.Component<ReactAvailableVie
             outputDescription = output.description;
         }
         let traceContainerClassName = 'outputs-list-container';
-        if (props.index === this._lastSelectedOutputIndex) {
+        if (props.index === this.state.lastSelectedOutputIndex) {
             traceContainerClassName = traceContainerClassName + ' theia-mod-selected';
         }
         return <div className={traceContainerClassName}
-            title={outputName + ': ' + outputDescription}
```
Comment (PatrickTasse) [inline packages/react-components/src/trace-explorer/trace-explorer-views-widget.tsx]: If this was added by mistake in previous commit, that commit should be amended to remove it.

File: packages/react-components/style/trace-explorer.css [hunk 4]
```diff
@@ -74,16 +74,12 @@
 }
 
 .trace-element-name, .outputs-element-name {
-    overflow: hidden;
-    text-overflow: ellipsis;
```
Comment (PatrickTasse) [inline packages/react-components/style/trace-explorer.css]: These changes seem related and necessary to the first commit, but the first commit is unrelated to this PR.

If you intended to make these changes in a separate PR, the first commit and this commit should simply be removed from this PR.

**Review by PatrickTasse — CHANGES_REQUESTED** [review]

**Review by PatrickTasse — APPROVED** [review]

**Review by bhufmann — APPROVED** [review]
LGTM. Thanks for the contribution!

**Pull request merged**